src/components/productrecommendation.fetchallpages.jsx

```diff
@@ -0,0 +1,163 @@
+/**
+ * Copyright © 2018 Elastic Path Software Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ *
+ */
+
+import React from 'react';
+import PropTypes from 'prop-types';
+import { login } from '../utils/AuthService';
+import ProductListItemMain from './productlistitem.main';
+
+const Config = require('Config');
+
+
+class Productrecommendationfetchallpages extends React.Component {
+  static propTypes = {
+    product: PropTypes.objectOf(PropTypes.any).isRequired,
+    length: PropTypes.number.isRequired,
+    count: PropTypes.number.isRequired,
+  }
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      links: [],
+    };
+  }
+
+  componentDidMount() {
+    const { product } = this.props;
+    const { links } = this.state;
+    const link = links;
+    const arr = product.links;
+    for (let i = 0; i < arr.length; i += 1) {
+      if (arr[i].rel === 'element') {
+        link.push(arr[i]);
+      } else if (arr[i].rel === 'next') {
```

Comment:
this seems a bit silly, you're taking paginated results from cortex, stringing them up into one big list, and then you divide it up again? avoids the purpose of the pagination doesn't it? the max items in one carousel should be the page-size from cortex. and it should be customizable in CM already. adding this new configuration for the size of the carousel adds some logic that really isn't needed.

Reply:
I agree 100%. Recommendations the way they are designed in cortex.. i doubt they were ever intended to load in a carousel. The whole requirement of having a carousel with multi items  is a major work on front end. Nothing is out of the box. neither for cortex nor for bootstrap.
Cortex paginates the recommendations.. So have to pull all first.
Bootstrap 4 does not support multi item carousel. So have to hack it in there.